inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -257,11 +777,20 @@ std::pair<bool, std::string> compare_functions(
                         << std::endl;
             }
         }
-    }
 
+        if (compareAttributes) {
+            CompareNodesAttributes compare_nodes;
+            node1->visit_attributes(compare_nodes.get_ref_reder());
+            node2->visit_attributes(compare_nodes.get_cmp_reader());
```

Comment:
I'm not sure, but shouldn't we call validate_and_infer_type after visit_attributes?
(something related to default values for arguments was mentioned in prev reviews, maybe different context)

Reply:
This visitor is only reading values. Nothing is (re)set so I my opinion we don't have to.